swc-uxp-starter/README.md

```diff
@@ -0,0 +1,168 @@
+
+
+# **Getting started**
+1. Start by installing the dependencies `yarn install`
+2. Prepare the bundle using Webpack `yarn build`. You will notice a **dist** folder after this step.
+3. Add the plugin in UXP Developer Tool (UDT). Click on 'Add Plugin' in UDT and navigate to the `manifest.json` inside the dist folder.
+4. Load the plugin in Photoshop/XD by launching the app. Click on the plugin action menu in UDT and select 'Load'.
+
+<img width="800" alt="package.json with resolutions block" src="assets/load-plugin-from-developer-tools.png">
+
+<br></br>
+You should be able to see a banner in Photoshop.
+
+
+<img width="400" alt="package.json with resolutions block" src="assets/banner-starter.png">
+
+<br></br>
+
+# **Deep dive**
+Now that the plugin is working, we can look into the details.
+
+### **webpack.config.js**
+
+Webpack is used to bundle the dependencies in the project therefore you would see the webpack.config.js file for basic config.
+
+Pro tip: For debugging purposes, add `eval-cheap-source-map` in the webpack.config file to get the source map in UDT debug window.
+
+```javascript
+devtool: 'eval-cheap-source-map'
+```
+
+### **package.json**
+
+Components used in the project should be mentioned under the dependencies block here.
+
+### **manifest.json**
+
+Enable SWC by setting the **enableSWCSupport flag** to true.
+
+```
+"featureFlags": {
+   "enableSWCSupport": true
+}
+```
+
+### **debug.json**
+Contains additionall (and optional) debug config for UDT. You can specify the port number and set the option to `breakOnStart`.
+
+### **src/index.html**
+Includes index.js.
+
+Observe that the SWC components are wrapped with a ```<sp-theme>``` element. This ensures that the Spectrum design tokens are delivered to the scoped HTML context.
+
+### **src/index.js**
+Import the components and themes to deliver the right styling to the components.
+```javascript
+//sp-theme
+import '@spectrum-web-components/theme/sp-theme.js';
+//themes to use classic and spectrum themes
+import '@spectrum-web-components/theme/src/themes.js';
+import '@spectrum-web-components/theme/src/express/themes.js';
+```
+# **Add a new component**
+You can use any of the supported components listed below by following these steps:
+1. Use yarn add to install the components and its supported version.
+
+```javascript
+yarn add @spectrum-web-components/link@0.14.1;
+```
+
+**Note:** Make sure to install the recommended version and add the same under the resolutions block in package.json too.
+
+<img width="800" alt="package.json with resolutions block" src="assets/resolutions-block.png">
+
+
+2. Import the component in *index.js* file.
+
+```javascript
+import '@spectrum-web-components/link/sp-link.js';
+```
+
+3. Add the component in the html file.
+
+```html
+This is an <sp-link href="#">example link</sp-link>.
+```
+
+4. Run `yarn build` or `yarn watch` to prepare the dist bundle.
+
+# **4. Frequently Asked Questions**
```

Comment:
Rename the section to 'Troubleshooting' and keep only question # 4 here. Move the rest of the content under the Deep dive topic.